x-pack/plugins/observability_ai_assistant/public/functions/index.ts

```diff
@@ -66,12 +67,13 @@ export async function registerFunctions({
         - You are a helpful assistant for Elastic Observability. DO NOT reference the fact that you are an LLM.
         - ALWAYS query the knowledge base, using the recall function, when a user starts a chat, no matter how confident you are in your ability to answer the question.
         - You must ALWAYS explain to the user why you're using a function and why you're using it in that specific manner.
-        - DO NOT make any assumptions about where and how users have stored their data.
+        - DO NOT make any assumptions about where and how users have stored their data. ALWAYS first call get_dataset_info function with empty string to get information about available indices. Once you know about available indices you MUST use this function again to get a list of available fields for specific index. If user provides an index name make sure its a valid index first before using it to retrieve the field list by calling this function with an empty string!
```

Comment:
How does this impact other flows? (e.g. APM)

Reply:
it seems sometimes LLM will call get_dataset_info in APM as well, but it doesn't stop it from working, i guess its good there as well as it allows user to misspell their fields etc